### PR TITLE
[go-gen] Add update handler generation

### PR DESCRIPTION
**Pull request by willjvsmith**
* Adds update handler generation
  * Should handle readable/writable/auth/project-auth cases

Almost all of the blocks are already generated for the other handlers, so was simply a case of composing them 👌

### CODE REVIEW

File: src/e2e/resources/simple-temple-expected/simple-temple-test-group/simple-temple-test-group.go [hunk 1]
```diff
@@ -189,7 +189,75 @@ func (env *env) readSimpleTempleTestGroupHandler(w http.ResponseWriter, r *http.
...
+
+	var req updateSimpleTempleTestGroupRequest
+	err = json.NewDecoder(r.Body).Decode(&req)
+	if err != nil {
+		errMsg := util.CreateErrorJSON(fmt.Sprintf("Invalid request parameters: %s", err.Error()))
+		http.Error(w, errMsg, http.StatusBadRequest)
+		return
+	}
+
+	if {
```
Comment (jaylees14) [inline src/e2e/resources/simple-temple-expected/simple-temple-test-group/simple-temple-test-group.go]: wat

**Review by jaylees14 — CHANGES_REQUESTED** [review]
Copying and pasting? ;)

File: src/e2e/resources/simple-temple-expected/simple-temple-test-group/simple-temple-test-group.go (hunk 1 above)
Comment (willjvsmith) [reply src/e2e/resources/simple-temple-expected/simple-temple-test-group/simple-temple-test-group.go]: Ah this is part of the `Handle the case of no user-defined/client-facing attributes` card, will be handled in another PR

Comment (jaylees14) [reply src/e2e/resources/simple-temple-expected/simple-temple-test-group/simple-temple-test-group.go]: Can you at least make it valid Go? `if false`?

**Review by jaylees14 — APPROVED** [review]

File: src/main/scala/temple/generate/server/go/service/main/GoServiceMainCreateHandlerGenerator.scala [hunk 2]
```diff
@@ -74,7 +67,8 @@ object GoServiceMainCreateHandlerGenerator {
         mkCode.doubleLines(
           when(root.projectUsesAuth) { generateExtractAuthBlock(usesVar = true) },
           generateDecodeRequestBlock(s"create${root.name}"),
-          generateRequestNilCheck(root, clientAttributes),
+          // TODO: Handle this properly, there could be serverSet attributes
+          when(clientAttributes.nonEmpty) { generateRequestNilCheck(root, clientAttributes) },
```
Comment (c-harding) [inline src/main/scala/temple/generate/server/go/service/main/GoServiceMainCreateHandlerGenerator.scala:71]: What’s happening with this TODO?

File: src/main/scala/temple/generate/server/go/service/main/GoServiceMainUpdateHandlerGenerator.scala [hunk 3]
```diff
@@ -1,16 +1,54 @@
...
       generateHandlerDecl(root, Update),
-      CodeWrap.curly.tabbed(),
+      CodeWrap.curly.tabbed(
+        mkCode.doubleLines(
+          when(root.projectUsesAuth) { generateExtractAuthBlock(root.writable == Writable.This) },
+          generateExtractIDBlock(root.decapitalizedName),
+          when(root.writable == Writable.This) { generateCheckAuthorizationBlock(root) },
+          generateDecodeRequestBlock(s"update${root.name}"),
+          // TODO: Handle this properly, there could be serverSet attributes
+          when(clientAttributes.nonEmpty) { generateRequestNilCheck(root, clientAttributes) },
```
Comment (c-harding) [inline src/main/scala/temple/generate/server/go/service/main/GoServiceMainUpdateHandlerGenerator.scala:46]: ditto

**Review by c-harding — APPROVED** [review]
nice

File: src/main/scala/temple/generate/server/go/service/main/GoServiceMainCreateHandlerGenerator.scala (hunk 2 above)
Comment (willjvsmith) [reply src/main/scala/temple/generate/server/go/service/main/GoServiceMainCreateHandlerGenerator.scala:71]: See comments in Jay's review: https://github.com/TempleEight/temple/pull/253#discussion_r404765886

This is a temporary fix to make the generated Go valid

**Pull request merged**